### PR TITLE
Space separator

### LINKED ISSUE #5217: Invitations - Remove space bar as confirmation
url: https://staging.archipelmarket.com/c/o/dashboard/event/h91BbmBH9OMDnazNLwi3/edit
user dev+brandyn-xio@blockframes.io

When writing emails or name to invite people, the space bar is considered as confirmation. It's really annoying as you can search someone by his/her name so you want to be able to write both first name, space bar, last name. 
This space bar behaviour should simply be removed.

### PR DESCRIPTION
**Pull request by Mussuranae**
Fix #5217

### CODE REVIEW

File: libs/invitation/src/lib/form/user/user.component.ts [hunk 1]
```diff
@@ -82,9 +82,20 @@ export class UserComponent implements OnInit {
       const fromOrg = this.ownerOrgId ? await this.orgService.getValue(this.ownerOrgId) : undefined;
       await this.invitationService.invite(emails, fromOrg).to('attendEvent', this.eventId);
       this.sending.next(false);
+      /*
+      Remove the Space separator when the user invites the list of emails.
+      If we don't do that, the user won't be able to search after people because
+      the space bar will generate a new chip that will be considered as a new email
+      */
+      if (this.separators.includes(SPACE)) this.separators.splice(this.separators.length -1, 1);
```
Comment (RemcoSimonides) [inline libs/invitation/src/lib/form/user/user.component.ts:90]: Removing last element from array can be done with `this.separators.pop()` 
But if we check if the array includes SPACE, we should also check it position in my opinion.
```
const index = this.separators.indexOf(separator => separator === SPACE);
if (index > -1) this.separators.splice(index, 1)
```

**Review by RemcoSimonides — APPROVED** [review]

File: libs/invitation/src/lib/form/user/user.component.ts [hunk 2]
```diff
@@ -82,9 +82,15 @@ export class UserComponent implements OnInit {
...
       await this.invitationService.invite(emails, fromOrg).to('attendEvent', this.eventId);
       this.sending.next(false);
+      this.separators.splice(this.separators.length -1, 1);
     }
   }
 
+  /** Add the SPACE separator if the user pastes email addresses */
+  onPaste(event: ClipboardEvent) {
+    if (event.type === 'paste') this.separators.push(SPACE);
+  }
```
[the remark this answers is not in the document]
Comment (Mussuranae) [reply libs/invitation/src/lib/form/user/user.component.ts:97]: I can't do it in the same function (or at least, I don't know how :/), that's why I put it in the invite function. So the user will have to validate these pasted emails and send a first wave of invitations, and after that he can type and search for users he wants to invite.

[the remark this answers is not in the document]
Comment (Mussuranae) [reply libs/invitation/src/lib/form/user/user.component.ts:97]: Let me try some things

**Review by RemcoSimonides — APPROVED** [review]
Use `{}` for the if statement too. I think it's more neat since it's not a one-liner now anyway

**Pull request merged**